---
title: "02_clean_data"
format:
  html:
    embed-resources: true
knitr:
  opts_chunk:
    dev: "svg"
    fig.path: "../results/"
editor: visual
---

# Load libraries

```{r}
#| label: setup
#| message: false
library("tidyverse")
```

# Clean Data

## Load data

```{r}
#| label: Read raw data
meta_data <- read.csv("../data/01_data_table_load.csv", na = c("", "NA", "N/A"))
disease_label <- read.csv("../data/01_disease_state_vector_load.csv")
```

## Select relevant columns and remove all NAs

Meta data

```{r}
selected_data <- meta_data |>
  select(-Gene.symbol, -Gene.ID, -UniGene.title, -UniGene.symbol, -UniGene.ID, 
         -Nucleotide.Title, -GI, -GenBank.Accession, -Platform_CLONEID, - Platform_ORF,
         -Platform_SPOTID, -Chromosome.location, -Chromosome.annotation, -X) |>
  drop_na()
```

Disease Label data

```{r}
disease_label <- disease_label |>
  select(-X)
```

## Log2 transformation of expression levels

```{r}
transformed_data <- selected_data |>
  mutate(across(starts_with("GSM"), log2))
```

## Remove duplicates

Some genes (IDENTIFIER) have been measured with multiple probes (ID_ref).

We selected the probe with the highest median expression for each gene, as this provides the most robust and reliably measured representation of gene expression levels.

```{r}
# Aggregate multiple probes on same gene: 
# keep the probe with highest median expression for each gene
aggregated_data <- transformed_data |>
  group_by(IDENTIFIER) |>
  mutate(probe_median = apply(across(starts_with("GSM")), 1, median)) |>
  slice_max(probe_median, n = 1, with_ties = FALSE) |>  # Keep the best-expressed probe
  select(-probe_median) |>
  ungroup()
```

## Merge data

The transformed data is merged with the disease_label data. The label is merged by "Sample" and is added as the last row in the data set.

This data set has subjects as columns and genes as observations.

```{r}
label_row <- disease_label |> 
  pivot_wider(
    names_from = Sample,
    values_from = disease.state) |> 
  mutate(
    IDENTIFIER = "disease.state") |> 
  select(IDENTIFIER, everything())

data_as_characters <- aggregated_data |> 
  mutate(across(starts_with("GSM"), as.character))

df_clean <- bind_rows(data_as_characters, label_row)
```

## Save clean data as a csv-file.

```{r}
write.csv(df_clean, file="../data/02_clean_data.csv")
```

# Transpose Data

For some parts of the further analysis,

```{r}
df_clean_selected <- df_clean |> 
  select(IDENTIFIER, starts_with("GSM"))

df_transposed <- as.data.frame(t(df_clean_selected))

df_transposed_clean <- df_transposed |> 
  rename_with(~ as.character(slice(df_transposed, 1) |> 
                              unlist())) |> 
  slice(-1)
```

Save transposed data as a csv file

```{r}
write.table(df_transposed_clean, file="../data/02_transposed_data.csv")
```
